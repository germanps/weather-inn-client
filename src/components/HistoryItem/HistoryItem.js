import React from 'react'
import { useFetch } from '../../hooks/useFetch'
import Spinner from '../Spinner';
import { getItemIcon } from '../../utils/utils'
import WeatherIcons from 'react-weathericons'
import PropTypes from 'prop-types'
import './HistoryItem.scss'

export default function HistoryItems({ searchItem }) {
    const { loading, data } = useFetch(`provincias/${searchItem.codprov}/municipios/${searchItem.idpob}`)
    return (
        <>
            {
                loading ?
                    (
                        <Spinner />
                    ) :
                    (
                        <li className="history-item">
                            <div className="history-item__wrapper">
                                <span className="history-item__wrapper--text city">{data.municipio.NOMBRE}  {data.temperatura_actual}º</span>


                                <span className="history-item__wrapper--text degrees">
                                    <span className="history-item__wrapper--icon"><WeatherIcons name={getItemIcon(data.stateSky.description)} size="2x" /></span>
                                    <span className="max">{`Máx: ${data.temperaturas.max}º`}</span>
                                    <span className="min">{`Mín: ${data.temperaturas.min}º`}</span>
                                </span>
                            </div>
                        </li>
                    )
            }

        </>
    )
}


HistoryItems.propTypes = {
    weatherToday: PropTypes.object,
}


