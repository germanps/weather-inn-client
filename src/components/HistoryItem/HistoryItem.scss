@import "../../scss/index.scss";

.history-item{
    font-size: $font-size * .8;
    background-color:$action;
    margin-bottom: .2rem;
    padding: .65rem;
    box-sizing: border-box;
    max-width: 400px;
    box-shadow: 0 2px 2px -1px rgba(152, 162, 179, 0.3), 0 1px 5px -2px rgba(152, 162, 179, 0.3);
    border: 1px solid $action;
    
    &__wrapper{
        @include flex(100%, column, flex-start, flex-start);
        @media(min-width: $media-small) {
            flex-direction: row;
            justify-content: space-between;
            align-items: center;
        }
        &--text{
            width: 180px;
            max-width: 100%;
            color: $font-light;
            
            &.city{
                margin-bottom: .5rem;
                @media(min-width: $media-small) {
                    width: calc(100% - 140px);
                }
                @media(min-width: $media-small) {
                    margin-bottom: 0;
                }
            }
            &.degrees{
                @include flex(140px, row, space-between, center);
            }
        }
        &--icon{
            margin-left: 0;
            i{
                font-size: 1rem !important;
                color: $font-light;
            }
        }
    }
}