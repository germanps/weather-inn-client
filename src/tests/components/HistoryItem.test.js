import React from 'react'
import { shallow } from 'enzyme'
import PropTypes from 'prop-types'
import HistoryItem from '../../components/HistoryItem'

describe('Test in <HistoryItem />', () => {
    const searchItem = { codprov: 'prov', idpob: 'pob' }
    test('should show the component correctly', () => {
        const wrapper = shallow(<HistoryItem searchItem={searchItem} />)
        expect(wrapper).toMatchSnapshot()
    })

})

HistoryItem.propTypes = {
    weatherToday: PropTypes.object,
}