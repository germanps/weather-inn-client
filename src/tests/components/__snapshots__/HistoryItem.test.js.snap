// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <HistoryItem /> should show the component correctly 1`] = `
<Fragment>
  <Spinner />
</Fragment>
`;
